status/api/accounts.nim

```diff
@@ -0,0 +1,187 @@
+{.push raises: [Defect].}
+
+import # std libs
+  std/[json, tables, times]
+
+import # vendor libs
+  eth/keyfile/uuid, secp256k1
+
+import # nim-status libs
+  ../alias, ../accounts/[accounts, generator/generator, public_accounts],
+  ./common, ../conversions, ../extkeys/[paths, types], ../identicon, ../settings
+
+export
+  accounts, alias, common, conversions, generator, identicon, paths,
+  public_accounts, secp256k1, settings, types, uuid
+
+type
+  PublicAccountResult* = Result[PublicAccount, string]
+
+proc storeDerivedAccounts(self: StatusObject, id: UUID, keyUid: string,
```

Comment:
In our api modules, would it make sense to attempt to only include procs that are public/exported?

Thinking about it, I see how there can arise some tricky situations re: circular dependencies. Also, at times it wouldn't be worth the effort to refactor a non-exported helper into a non-api module (from which it would be exported, but not re-exported by the api module).

In a lot of cases, maybe it would be a worthwhile exercise to at least consider whether a non-exported proc could be moved and what kinds of entanglements are involved. Or maybe not... needs more consideration.

Reply:
I understand what you're asking and definitely agree it needs more consideration in general. In this particular instance, `storeDerivedAccounts` is a specific abstraction that hides implementation details from the consumer. As mentioned in https://github.com/status-im/nim-status/pull/254#discussion_r681484712, highly motivated consumers could import all not re-exported modules explicitly and roll their own implementation of `storeDerivedAccounts` if so desired.